pkg/kubelet/network/dns/dns.go

```diff
@@ -63,20 +66,50 @@ type Configurer struct {
 	// the container's DNS resolver configuration file. This can be used in
 	// conjunction with clusterDomain and clusterDNS.
 	ResolverConfig string
+
+	// client to access network CR
+	arktosV1Client arktosv1.ArktosV1Interface
 }
 
 // NewConfigurer returns a DNS configurer for launching pods.
-func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, nodeIP net.IP, clusterDNS []net.IP, clusterDomain, resolverConfig string) *Configurer {
+func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, nodeIP net.IP, clusterDNS []net.IP, clusterDomain, resolverConfig string, arktosV1Client arktosv1.ArktosV1Interface) *Configurer {
 	return &Configurer{
 		recorder:       recorder,
 		nodeRef:        nodeRef,
 		nodeIP:         nodeIP,
 		clusterDNS:     clusterDNS,
 		ClusterDomain:  clusterDomain,
 		ResolverConfig: resolverConfig,
+		arktosV1Client: arktosV1Client,
 	}
 }
 
+func (c *Configurer) getClusterDNS(pod *v1.Pod) []net.IP {
+	const defaultNetwork = "default"
+	networkName, ok := pod.Labels["arktos.futurewei.com/network"]
+	if !ok {
+		networkName = defaultNetwork
+	}
+	if network, err := c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant).Get(networkName, metav1.GetOptions{}); err == nil {
```

Comment:
Needs to check if c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant).Get() will trigger nil pointer exception if the network CRD is not installed.

Reply:
c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant) won't return nil; it just sets up tenant value as client property, which will be used to assemble the rest path of network CR's url. So no nil reference risk here.